src/psm/src/ir_solver.cpp

```diff
@@ -536,30 +534,65 @@ bool IRSolver::CreateGmat(bool connection_only) {
             via_bottom_layer = via->getBottomLayer();
           } else {
             dbTechVia* via = curWire->getTechVia();
-            via_bBox = via->getBBox();
             check_params = via->hasParams();
             if (check_params) {
               via->getViaParams(params);
             }
             via_top_layer = via->getTopLayer();
             via_bottom_layer = via->getBottomLayer();
           }
-          BBox bBox =
-              make_pair((via_bBox->getDX()) / 2, (via_bBox->getDY()) / 2);
+          int x_cut_size = 0;
+          int y_cut_size = 0;
+          int x_bottom_enclosure = 0;
+          int y_bottom_enclosure = 0;
+          int x_top_enclosure = 0;
+          int y_top_enclosure = 0;
```

Comment:
If the via doesn't have parameters wouldn't you want to get the sizes from the via's boxes?  Even if it has parameters the boxes be simpler to use.  This applies elsewhere as well.

Reply:
I'm not sure if you were referring to bbox or to via_boxes. I have used the former as it appeared to be a more direct solution. I still need the params for number of cuts for accurate resistance calc.